### ISSUE TITLE
missing typescript overload for cursor.eachasync

### ISSUE THREAD

**Issue by krosenk729** [Automattic/mongoose#13152]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.6.1

### Node.js version

18+

### MongoDB server version

6+

### Typescript version (if applicable)

4.5.5

### Description

The `fn` callback for eachAsync should allow for an index parameter; however, the typescript definitions only allow for a single DocType or DocType[] param.

The function is properly shown in the documentation to allow `i` as an index here https://mongoosejs.com/docs/api/querycursor.html#querycursor_QueryCursor-eachAsync but the typescript definitions below prevent use of the second arg

```
    eachAsync(fn: (doc: DocType[]) => any, options: EachAsyncOptions & { batchSize: number }, callback: CallbackWithoutResult): void;
    eachAsync(fn: (doc: DocType) => any, options: EachAsyncOptions, callback: CallbackWithoutResult): void;
    eachAsync(fn: (doc: DocType[]) => any, options: EachAsyncOptions & { batchSize: number }): Promise<void>;
    eachAsync(fn: (doc: DocType) => any, options?: EachAsyncOptions): Promise<void>;
```

### Steps to Reproduce

Use mongoose in a `.ts` file and try `Thing.find({}).cursor().eachAsync(async (doc, i) => { })` ... this will cause a typescript error `No overload matches this call.`

**Comment by hasezoey** [comment]
closing because #13153 as been merged and will be available in 7.0.2

**Comment by hasezoey** [comment]
nvm, forgot this issue was for 6.x and 7.x
